### ISSUE TITLE
[MRESOLVER-91] Error messages not clear when not reattempting resolution when a previous attempt failed and the failure was cached

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resolver#829]
**[Giorgos Gaganis](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gaganis)** opened **[MRESOLVER-91](https://issues.apache.org/jira/browse/MRESOLVER-91?redirect=false)** and commented

Error messages not clear when not reattempting resolution when a previous attempt  failed and the failure was cached.

We have experienced the following error message 

```java
Failure to find groupid:artifactid:0.0.1 in http://therepo/path/repo was cached in the local repository, resolution will not be reattempted until the update interval of TheRepo has elapsed or updates are forced -> [Help 1]
```

At the time we have found this error message to be confusing, we couldn't really understand what was happening. In our case while we had a simple error in our repo configuration we were misled by this message and looking for the problem in our local repository caching.

During jcrete hackday I was still confused by this but decided to look into it and this led me to the code where I found the description of the mechanism in org.eclipse.aether.resolution.ResolutionErrorPolicy. After reading the javadoc and understanding the mechanism actually the error message makes absolute sense.

So we might be able to improve the error message so it conveys better what is happeing for users seeing this for the first time and not being aware or understading the failure caching mechanism.

I will attach a PR with a recommendation for better error messages. With this wording we convey better what has happened before, what exactly was cached from before and why we are currently failing(by not reattempting).

---
No further details from [MRESOLVER-91](https://issues.apache.org/jira/browse/MRESOLVER-91?redirect=false)

**Comment by jira-importer** [comment]
**[Giorgos Gaganis](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gaganis)** commented

Proposed PR https://github.com/apache/maven-resolver/issues/37

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [55fa2d7e594afebfde4a7e1d3cec23a2efc63d1d](https://gitbox.apache.org/repos/asf?p=maven-resolver.git&a=commit&h=55fa2d7e594afebfde4a7e1d3cec23a2efc63d1d).
